examples/models/llama2/runner/runner.cpp

```diff
@@ -204,16 +205,20 @@ Error Runner::generate(
   // after the prompt. After that we will enter generate loop.
 
   // print prompts
-  wrapped_callback(prompt);
-
+  if (echo) {
+    wrapped_callback(prompt);
+  }
+  
   auto prefill_res = text_prefiller_->prefill(prompt_tokens, 0);
   stats_.first_token_ms = util::time_in_ms();
   stats_.prompt_eval_end_ms = util::time_in_ms();
   ET_CHECK_OK_OR_RETURN_ERROR(prefill_res.error());
   uint64_t cur_token = prefill_res.get();
 
   // print the first token from prefill. No prev_token so use cur_token for it.
-  wrapped_callback(ET_UNWRAP(tokenizer_->decode(cur_token, cur_token)));
+  if (echo) {
+    wrapped_callback(ET_UNWRAP(tokenizer_->decode(cur_token, cur_token)));
+  }
```

Comment:
I think this is needed, otherwise we are missing the first generated token.

Reply:
from my testing, this callback would just return a newline. We also tested this and the output works well.

Reply:
Hmm I'm saying it doesn't matter what that token is. Semantically it is the first token the model generates, by prefilling the prompt, so we should trigger callback on this one.

Reply:
got it! @larryliu0820, I've removed the echo around the first token generation.